projects/RabbitMQ.Client/client/api/ConnectionFactory.cs

```diff
@@ -410,6 +411,19 @@ public IConnection CreateConnection()
             return CreateConnection(ClientProvidedName);
         }
 
+        /// <summary>
+        /// Asynchronously reate a connection to one of the endpoints provided by the IEndpointResolver
+        /// returned by the EndpointResolverFactory. By default the configured
+        /// hostname and port are used.
+        /// </summary>
+        /// <exception cref="BrokerUnreachableException">
+        /// When the configured hostname was not reachable.
+        /// </exception>
+        public ValueTask<IConnection> CreateConnectionAsync()
```

Comment:
That is also a more general comment: where is the support for cooperative cancelation? Will it come in a separate PR?

Reply:
Yes, because this PR has gone on too long. I will open an issue to more generally support cancellation. Of course, this won't actually cancel anything RabbitMQ-side.

Reply:
https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1420